integration/kubernetes/confidential/image_pulling_with_snapshotter.bats

```diff
@@ -0,0 +1,160 @@
+#!/usr/bin/env bats
+# Copyright (c) 2023 Intel Corporation
+# Copyright (c) 2023 IBM Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+load "${BATS_TEST_DIRNAME}/lib.sh"
+load "${BATS_TEST_DIRNAME}/../../confidential/lib.sh"
+
+tag_suffix=""
+if [ "$(uname -m)" != "x86_64" ]; then
+    tag_suffix="-$(uname -m)"
+fi
+
+# Images used on the tests.
+
+image_unsigned_protected="quay.io/kata-containers/confidential-containers:unsigned${tag_suffix}"
+
+original_kernel_params=$(get_kernel_params)
+# Allow to configure the runtimeClassName on pod configuration.
+RUNTIMECLASS="${RUNTIMECLASS:-kata}"
+test_tag="[cc][agent][kubernetes][containerd]"
+
+# Create the test pod.
+#
+# Note: the global $sandbox_name, $pod_config should be set
+# 	already. It also relies on $CI and $DEBUG exported by CI scripts or
+# 	the developer, to decide how to set debug flags.
+#
+create_test_pod() {
+    local pod_config="$1"
+    # On CI mode we only want to enable the agent debug for the case of
+    # the test failure to obtain logs.
+    if [ "${CI:-}" == "true" ]; then
+        enable_full_debug
+    elif [ "${DEBUG:-}" == "true" ]; then
+        enable_full_debug
+        enable_agent_console
+    fi
+
+    echo "Create the test sandbox"
+    echo "Pod config is: $pod_config"
+    kubernetes_create_cc_pod $pod_config
+}
+
+# Create a pod configuration out of a template file.
+#
+# Parameters:
+#	$1 - the container image.
+# Return:
+# 	the path to the configuration file. The caller should not care about
+# 	its removal afterwards as it is created under the bats temporary
+# 	directory.
+#
+# Environment variables:
+#	RUNTIMECLASS: set the runtimeClassName value from $RUNTIMECLASS.
+#
+new_pod_config() {
+    local base_config="${FIXTURES_DIR}/pod-config.yaml.in"
+    local image="$1"
+    local index="$2"
+
+    local new_config=$(mktemp "${BATS_FILE_TMPDIR}/$(basename ${base_config}).XXX")
+    IMAGE="$image" RUNTIMECLASS="$RUNTIMECLASS" INDEX="$2" envsubst <"$base_config" >"$new_config"
+    echo "$new_config"
+}
```

Comment:
I think this code is copy-pasted from agent_image.bats, so maybe you could move it to `confidential/lib.sh` and then call it from both bats files?

Reply:
Sure. I'll update it.